rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -574,6 +574,12 @@ class Publisher : public PublisherBase
   std::unique_ptr<PublishedType, PublishedTypeDeleter>
   duplicate_type_adapt_message_as_unique_ptr(const PublishedType & msg)
   {
+    /// Assert that the published type has no overloaded operator new since this leads to
+    /// new/delete mismatch (see https://github.com/ros2/rclcpp/issues/2951)
+    static_assert(!detail::has_overloaded_operator_new_v<PublishedType>,
+        "When publishing by value (i.e. when calling publish(const T& msg)), the published message"
+        "type must not have an overloaded operator new. In this case, please use the"
```

Comment:
Minor suggestion. Please double-check the line length.

```suggestion
        "When publishing by value (i.e. when calling publish(const T& msg)), the published message "
        "type must not have an overloaded operator new. In this case, please use the "
```

Reply:
I've used now  'ament_uncrustify --reformat --linelength 100` for code formatting, which formatter do you use ?